File: frontend/src/lib/productLexicon.ts
/** Display labels for product kinds. Canonical keys stay English. */

export const PRODUCT_KIND_LABELS: Record<string, string> = {
  course_15: "Курс 15",
  main_course: "Курс",
  protocol: "Протокол",
  other_service: "Другая услуга",
  visit: "Визит",
  desk: "Desk",
  extra: "Доп. услуга",
};

/** main_course → «Курс»; otherwise catalog name or kind label. */
export function productDisplayLabel(
  kindOrKey: string | null | undefined,
  name?: string | null,
): string {
  const k = (kindOrKey || "").trim();
  const n = (name || "").trim();
  if (k === "main_course") return PRODUCT_KIND_LABELS.main_course;
  if (n) {
    if (k && PRODUCT_KIND_LABELS[k]) {
      const nl = n.toLowerCase();
      if (nl === k.toLowerCase() || nl === "основной курс" || nl === "main course") {
        return PRODUCT_KIND_LABELS[k];
      }
    }
    return n;
  }
  if (!k) return "—";
  return PRODUCT_KIND_LABELS[k] ?? k;
}

File: backend/app/services/product_lexicon.py
"""Display labels for product kinds — UI only; canonical kinds unchanged."""

from __future__ import annotations

# kind → пользовательское имя. Catalog KPI «Курс» ↔ kind main_course.
PRODUCT_KIND_LABELS: dict[str, str] = {
    "course_15": "Курс 15",
    "main_course": "Курс",
    "protocol": "Протокол",
    "other_service": "Другая услуга",
    "visit": "Визит",
    "desk": "Desk",
    "extra": "Доп. услуга",
}


def product_display_label(kind: str | None, name: str | None = None) -> str:
    """Prefer catalog name when present; map known kinds for generic/legacy names."""
    k = (kind or "").strip()
    n = (name or "").strip()
    if k == "main_course":
        # KPI item often «Курс»; always show «Курс» for this kind.
        return PRODUCT_KIND_LABELS["main_course"]
    if n:
        if k in PRODUCT_KIND_LABELS and n.casefold() in {
            k.casefold(),
            "main course",
            "основной курс",
            "курс",
        }:
            return PRODUCT_KIND_LABELS[k]
        return n
    if k in PRODUCT_KIND_LABELS:
        return PRODUCT_KIND_LABELS[k]
    return k or "—"
